lib/hanami/server.rb

```diff
@@ -49,11 +49,35 @@ def middleware
       mw = Hash.new { |e, m| e[m] = [] }
       mw["deployment"].concat([::Rack::ContentLength, ::Rack::CommonLogger])
       mw["development"].concat(mw["deployment"] + [::Rack::ShowExceptions, ::Rack::Lint])
+      mw["development"].push(::Shotgun::Static) if enable_shotgun_static?
       mw
     end
 
     private
 
+    # @since x.x.x
+    # @api private
+    def enable_shotgun_static?
+      code_reloading? && shotgun_available?
+    end
+
+    # @since x.x.x
+    # @api private
+    def shotgun_available?
+      begin
+        require 'shotgun'
+        true
+      rescue LoadError
+        false
+      end
```

Comment:
I don't like implicit begin blocks (because we always will need explicit begin blocks as well and i don't like two styles for the same thing) but the ruby styleguide prefers them. Whatever is more common and preferred in Hanami.